resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/KubernetesConf.scala

```diff
@@ -275,7 +275,7 @@ private[spark] object KubernetesConf {
         .toLowerCase(Locale.ROOT)
         .replaceAll("[^a-z0-9\\-]", "-")
         .replaceAll("-+", "-"),
-      "",
+      "-", // avoid using empty string, see https://issues.apache.org/jira/browse/LANG-1463
```

Comment:
It's a little confusing that LANG-1463 is marked as fixed at 3.10 and Apache Spark is using 3.12.
If LANG-1463 is not fixed yet, please mention correctly instead of saying `see ...`.

Reply:
Yes, LANG-1463 is fixed at 3.10 and Apache Spark is using 3.12. Removed the confusing comment. And I explained why I propose this change in PR description, 

> Strictly speaking, it's not an issue of Spark ...